express/dataset.py

```diff
@@ -0,0 +1,209 @@
+"""This module defines the FondantDataset class, which is a wrapper around the manifest.
+
+It also defines the FondantComponent class, which uses the FondantDataset class to manipulate data.
+"""
+
+from abc import abstractmethod
+import argparse
+import json
+from pathlib import Path
+
+import dask.dataframe as dd
+
+from express.component_spec import ExpressComponent, kubeflow2python_type
+from express.manifest import Manifest, Subset, Index
+
+
+class FondantDataset:
+    """Wrapper around the manifest to download and upload data into a specific framework.
+
+    Uses Dask Dataframes for the moment.
+    """
+
+    def __init__(self, manifest):
+        self.manifest = manifest
+
+    @property
+    def project_name(self):
+        return self.manifest.project_name
+
+    def _load_subset(self, name: str) -> dd.DataFrame:
+        # get subset from the manifest
+        subset = self.manifest.get_subset(name)
+        # get its location and fields
+        # TODO remove gcp prefix
+        location = "gcs://" + subset.location + ".parquet"
```

Comment:
I don't think the location is always a parquet file. Sometimes you need to refer to a parquet folder that contains all the different paritions

Reply:
Yes the prefix and suffix additions are added as a todo at the top of this PR